samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/DialogAndWelcomeBot.java

```diff
@@ -0,0 +1,66 @@
+package com.microsoft.bot.sample.core;
+
+import com.codepoetics.protonpack.collectors.CompletableFutures;
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.io.IOUtils;
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.lang3.StringUtils;
+import com.microsoft.bot.builder.*;
+import com.microsoft.bot.dialogs.Dialog;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.Attachment;
+import com.microsoft.bot.schema.ChannelAccount;
+import com.microsoft.bot.schema.Serialization;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class DialogAndWelcomeBot extends DialogBot {
+    /**
+     * Creates a DialogBot.
+     *
+     * @param withConversationState ConversationState to use in the bot
+     * @param withUserState         UserState to use
+     * @param withDialog            Param inheriting from Dialog class
+     */
+    public DialogAndWelcomeBot(ConversationState withConversationState, UserState withUserState, Dialog withDialog) {
+        super(withConversationState, withUserState, withDialog);
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMembersAdded(
+        List<ChannelAccount> membersAdded, TurnContext turnContext
+    ) {
+        return turnContext.getActivity().getMembersAdded().stream()
+            .filter(member -> !StringUtils
+                .equals(member.getId(), turnContext.getActivity().getRecipient().getId()))
+            .map(channel -> {
+                Attachment welcomeCard = createAdaptiveCardAttachment();
+                Activity response = MessageFactory.attachment(welcomeCard);
+
+                return turnContext.sendActivity(response).thenApply(sendResult -> {
+                    return Dialog.run(getDialog(), turnContext, getConversationState().createProperty("DialogState"));
+                });
+            })
+            .collect(CompletableFutures.toFutureList())
```

Comment:
Is this necessary? I'm not quite sure about this. I think that we need to check another examples of `onMembersAdded`

Reply:
Yes, without this the code would recieve a Stream instead a CompletableFuture so it wont compile, I used [this](https://github.com/southworks/botbuilder-java/blob/24dc4e99d6dd17a5e31334f6e70daa303e50084a/samples/06.using-cards/src/main/java/com/microsoft/bot/sample/usingcards/RichCardsBot.java#L44) as a reference

Reply:
Agree